sdk/core/azure-core-management/src/main/java/com/azure/core/management/AzureEnvironment.java

```diff
@@ -323,7 +332,9 @@ public enum Endpoint {
         /** Microsoft Graph APIs endpoint. */
         MICROSOFT_GRAPH("microsoftGraphResourceId"),
         /** Managed HSM DNS suffix. */
-        MANAGED_HSM("managedHsmDnsSuffix");
+        MANAGED_HSM("managedHsmDnsSuffix"),
+        /** Storage DNS suffix. */
+        STORAGE("storageEndpointSuffix");
```

Comment:
We need to provide a default value in major clouds

https://github.com/Azure/azure-sdk-for-java/blob/c26da695af442ddba0b6b3bb867d714c34ada29d/sdk/core/azure-core-management/src/main/java/com/azure/core/management/AzureEnvironment.java#L62-L90